lms/djangoapps/courseware/url_helpers.py

```diff
@@ -43,4 +46,8 @@ def get_redirect_url(course_key, usage_key):
             'courseware_position',
             args=(unicode(course_key), chapter, section, navigation_index(position))
         )
+
+    if final_target_id:
```

Comment:
When is this ever false? In my testing, whether I link to a section (`chapter`), subsection (`sequential`), unit (`vertical`), or block, `final_target_id` always is set to the ID of the block at the level requested, and it always adds `activate_block_id` to the query string.

```
/jump_to_id/fc66f15610ea46519c792665c3f16962  (section)
final_target_id: block-v1:BradenX+JUMP+TEST+type@chapter+block@fc66f15610ea46519c792665c3f16962
```